lib/src/github/api.rs

```diff
@@ -476,6 +483,67 @@ impl Session for GithubSession {
         Ok(pull_requests)
     }
 
+    async fn get_pull_requests_by_commit(
+        &self,
+        owner: &str,
+        repo: &str,
+        commit: &str,
+        head: Option<&str>,
+    ) -> Result<Vec<PullRequest>> {
+        let mut pull_requests = vec![];
+        let mut page = 1;
+
+        loop {
+            let mut next_prs: Vec<PullRequest> = match self
+                .client
+                .get::<Vec<PullRequest>>(&format!(
+                    "repos/{}/{}/commits/{}/pulls?head={}&per_page=100&page={}",
```

Comment:
this looks almost identical to the above method except for the request URL. i think better to split out a private method like `do_get_pull_requests` or something.

Reply:
also ping: thoughts on deduping here?